Scripts/Runtime/Jobs/JobTypes/AnvilJobBatchExtension.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Unity.Burst;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Jobs;
+using Unity.Jobs.LowLevel.Unsafe;
+using UnityEngine.Scripting;
+
+namespace Anvil.Unity.DOTS.Jobs
+{
+    public static class AnvilJobBatchExtension
+    {
+        //*************************************************************************************************************
+        // SCHEDULING
+        //*************************************************************************************************************
+        public static unsafe JobHandle ScheduleBatch<TJob>(this TJob jobData,
+                                                           int arrayLength,
+                                                           int minIndicesPerJobCount,
+                                                           JobHandle dependsOn = default)
+            where TJob : struct, IAnvilJobBatch
+        {
+            IntPtr reflectionData = WrapperJobProducer<TJob>.GetReflectionData();
+            WrapperJobStruct<TJob> wrapperData = new WrapperJobStruct<TJob>(ref jobData);
+
+
+            JobsUtility.JobScheduleParameters scheduleParameters = new JobsUtility.JobScheduleParameters(UnsafeUtility.AddressOf(ref wrapperData),
+                                                                                                         reflectionData,
+                                                                                                         dependsOn,
+                                                                                                         ScheduleMode.Parallel);
+
+            dependsOn = JobsUtility.ScheduleParallelFor(ref scheduleParameters, arrayLength, minIndicesPerJobCount);
+            return dependsOn;
+        }
+
+        //*************************************************************************************************************
+        // STATIC HELPERS
+        //*************************************************************************************************************
+
+        /// <summary>
+        /// This method is only to be called by automatically generated setup code.
+        /// Called by Unity automatically
+        /// </summary>
+        /// <typeparam name="TJob"></typeparam>
+        public static void EarlyJobInit<TJob>()
+            where TJob : struct, IAnvilJobBatch
+        {
+            WrapperJobProducer<TJob>.Initialize();
```

Comment:
Worth doing an assert on the stack for this to make sure there is only ever the one place that this is called?